modules/openapi-generator/src/main/resources/dart-dio/api.mustache

```diff
@@ -25,23 +25,38 @@ class {{classname}} {
 
             String path = "{{{path}}}"{{#pathParams}}.replaceAll("{" + "{{baseName}}" + "}", {{{paramName}}}.toString()){{/pathParams}};
 
-            // query params
-            Map<String, dynamic> queryParams = {};
-            Map<String, String> headerParams = Map.from(headers ?? {});
-            Map<String, String> formParams = {};
+        Map<String, dynamic> queryParams = {};
+        Map<String, String> headerParams = Map.from(headers ?? {});
+        dynamic bodyData;
 
         {{#headerParams}}
                 headerParams["{{baseName}}"] = {{paramName}};
         {{/headerParams}}
         {{#queryParams}}
                 queryParams["{{baseName}}"] = {{paramName}};
         {{/queryParams}}
-            queryParams.removeWhere((key, value) => value == null);
-            headerParams.removeWhere((key, value) => value == null);
-            formParams.removeWhere((key, value) => value == null);
+        queryParams.removeWhere((key, value) => value == null);
+        headerParams.removeWhere((key, value) => value == null);
 
-            List<String> contentTypes = [{{#consumes}}
-                "{{{mediaType}}}"{{#hasMore}},{{/hasMore}}{{/consumes}}];
+        List<String> contentTypes = [{{#consumes}}"{{{mediaType}}}"{{#hasMore}},{{/hasMore}}{{/consumes}}];
+
+
+        {{#formParams}}
+        Map<String, dynamic> formData = {};
```

Comment:
@athornz  If there are multiple form parameters, `Map<String, dynamic> formData = {};` will be generated in duplicate.

Reply:
ah yes good point